runtime/src/iree/hal/utils/debug_allocator.c

```diff
@@ -158,7 +158,8 @@ static iree_status_t iree_hal_debug_allocator_fill_on_device(
               IREE_HAL_QUEUE_AFFINITY_ANY, 1, &command, &command_buffer));
 
   iree_hal_semaphore_t* semaphore = NULL;
-  iree_status_t status = iree_hal_semaphore_create(device, 0ull, &semaphore);
+  iree_status_t status = iree_hal_semaphore_create(
+      device, 0ull, IREE_HAL_SEMAPHORE_FLAG_NONE, &semaphore);
```

Comment:
We may want to start calling out API changes like this in release notes / relnotes somehow.

* `iree_hal_semaphore_create` gained a `iree_hal_semaphore_flags_t` bitfield parameter. Existing call sites should be updated from `iree_hal_semaphore_create(device, initial_value, &out_semaphore)` to `iree_hal_semaphore_create(device, initial_value, IREE_HAL_SEMAPHORE_FLAG_NONE, &out_semaphore)`. Future changes will add flags for X, Y, and Z.